### DISCUSSION TITLE
get rid of `new Function('return this')()` in the final bundle

### DISCUSSION THREAD

**Discussion by taozhou-glean** [webpack/webpack#16734] [Q&A]
## Feature request

it looks like we still have `new Function('return this')()` in the final bundle even just target for web, i assume its from the global shim: https://github.com/webpack/webpack/blob/b67626c7b4ffed8737d195b27c8cea1e68d58134/lib/runtime/GlobalRuntimeModule.js#L19

![image](https://user-images.githubusercontent.com/121891361/219204465-4aba3af5-e959-49be-b962-57d2aadad275.png)

it will not cause any csp violation in most browsers as the path will not be executed, but when it does, it might hit CSP issues. I just saw this post (https://mathiasbynens.be/notes/globalthis) and it seems that we can get rid of the `new Function` for most browsers (except IE9, 10).

Wondering if it's possible we adapt that and makes an exception just for IE9,10 so if  IE9,10 not included in the `browserslist`, then the final bundle does not need the `new Function('return this')()`

**What is the expected behavior?**

no `new Function('return this')()` if target browser does not include IE9, 10

**What is motivation or use case for adding/changing the behavior?**

it's weird to see `new Function()` call when work on CSP, would be great if we can live without it.

**Comment by alexander-akait** [comment]
We have `new Function('return this')()` only in https://github.com/webpack/webpack/blob/main/lib/runtime/GlobalRuntimeModule.js and it is in `try`, so there is no problems, also we already check https://github.com/webpack/webpack/blob/main/lib/runtime/GlobalRuntimeModule.js#L23, yes can can reduce runtime to using `output.globalObject`(https://webpack.js.org/configuration/output/#outputglobalobject), but anyway we don't have such code for webpack v5 (if I am wrong please show me how you got such code), I will move it to dicussion, if we will found a place or a plugin I will reopen or create a new one

**Reply by taozhou-glean** [reply]
there is one problem with the `try catch`, even the violation is not visible in the console log, but the CSP-Report-Only will still send a violation reports  for that if the path is executed, that's also why we notice this ;) I will take a look at the `output.globalObject`, thanks for sharing the link

**Reply by ameyaapte1** [reply]
When using module federation we still have 'new Function('return this')()' is there any way to not generate this in production bundle?

**Reply by josdejong** [reply]
The solution to configure Webpack with `node: false` shared in the following StackOverflow issue worked for me: 

https://stackoverflow.com/questions/48695579/how-to-remove-eval-and-function-constructor-from-webpack-build-to-avoid-csp-issu
